app/services/process_stripe_payment.rb

```diff
@@ -0,0 +1,25 @@
+class ProcessStripePayment
+  attr_reader :controller, :amount, :email, :token
+  
+  def initialize(controller, amount, email, token)
```

Comment:
Yeah I wasn't massively happy about passing the controller but it was the only way I could get the flash message from the service object into the view.  I'm sure there's a better way, could you suggest any ideas?